pkgs/tools/graphics/goverlay/default.nix

```diff
@@ -0,0 +1,119 @@
+{ lib
+, writeScriptBin
+, bash
+, stdenv
+, fetchFromGitHub
+, fetchpatch
+, fpc
+, lazarus-qt
+, qt5
+, libX11
+, libqt5pas
+, coreutils
+, git
+, gnugrep
+, libnotify
+, mesa-demos
+, polkit
+, procps
+, systemd
+, vulkan-tools
+, which
+}:
+
+let
+  # Finds data files using the XDG Base Directory Specification
+  # See https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html
+  find-xdg-data-files = writeScriptBin "find-xdg-data-files" ''
+    #!${bash}/bin/sh
+    IFS=:
+    for xdg_data_dir in ''${XDG_DATA_HOME:-$HOME/.local/share}:''${XDG_DATA_DIRS:-/usr/local/share:/usr/share}; do
+      if [ -f "$xdg_data_dir/$1" ]; then
+        echo "$xdg_data_dir/$1"
+      fi
+    done
+  '';
+in stdenv.mkDerivation rec {
+  pname = "goverlay";
+  version = "0.5";
+
+  src = fetchFromGitHub {
+    owner = "benjamimgois";
+    repo = pname;
+    rev = version;
+    hash = "sha256-qS0GY2alUBfkmT20oegGpkhVkK+ZOUkJCPSV/wt0ZUA=";
+  };
+
+  patches = [
+    # Find replay-sorcery in PATH
+    # See https://github.com/benjamimgois/goverlay/pull/123
+    (fetchpatch {
+      url = "https://github.com/benjamimgois/goverlay/commit/09da4db26196f42578b11bd6541be5ede1125cdf.patch";
+      sha256 = "sha256-qYxAe5okKwRfmk7IBVq0nl6RUddKjge+TKzah+0VfeQ=";
+    })
+
+    # Support running previews without MangoHud
+    # See https://github.com/benjamimgois/goverlay/pull/124
+    (fetchpatch {
+      url = "https://github.com/benjamimgois/goverlay/commit/1a7e9eee5113c45e93000dfd41f79b36fd169475.patch";
+      sha256 = "sha256-qTRuULv/NLuZLdejC34JS1T+jKvdRyVQSzCBXtOaRME=";
+    })
+
+    # Find MangoHud & vkBasalt Vulkan layers using the XDG Base Directory Specification
```

Comment:
There is an issue for this upstream: https://github.com/benjamimgois/goverlay/issues/53, but the changes required upstream would be different and this workaround may not be accepted.